### PR TITLE
fix model builder limited container support for endpoint mode.

### PR DESCRIPTION
**Pull request by makungaj1**
… containers other than TGI and DJL

*Issue #, if available:*

*Description of changes:*
Fix: ModelBuilder limited container support for endpoint mode.

*Testing done:*
* unit tests
* integ tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/serve/builder/jumpstart_builder.py [hunk 1]
```diff
@@ -437,42 +432,52 @@ def _build_for_jumpstart(self):
...
+        if self.mode != Mode.SAGEMAKER_ENDPOINT:
+            if self._is_gated_model(self.pysdk_model):
+                raise ValueError(
+                    "JumpStart Gated Models are only supported in SAGEMAKER_ENDPOINT mode."
+                )
 
-        if "djl-inference" in image_uri:
-            logger.info("Building for DJL JumpStart Model ID...")
-            self.model_server = ModelServer.DJL_SERVING
+            if "djl-inference" in self.pysdk_model.image_uri:
```
Comment (qiyunz) [inline src/sagemaker/serve/builder/jumpstart_builder.py:448]: can we add `huggingface-pytorch-inference` in the image URI matching logic and map ti model server of MMS.

Comment (makungaj1) [reply src/sagemaker/serve/builder/jumpstart_builder.py:448]: Let's add it next as we may need to add tune capability for it.

Comment (qiyunz) [reply src/sagemaker/serve/builder/jumpstart_builder.py:448]: no tune function needed for the HF container, the goal is to make it work with local container mode first.

File: src/sagemaker/serve/builder/jumpstart_builder.py [hunk 2]
```diff
@@ -437,42 +468,58 @@ def _build_for_jumpstart(self):
...
+                self._build_for_tgi_jumpstart()
 
-            self.pysdk_model.tune = self.tune_for_djl_jumpstart
-        elif "tgi-inference" in image_uri:
-            logger.info("Building for TGI JumpStart Model ID...")
-            self.model_server = ModelServer.TGI
+                self.pysdk_model.tune = self.tune_for_tgi_jumpstart
+            elif "huggingface-pytorch-inference:" in self.pysdk_model.image_uri:
+                logger.info("Building for MMS JumpStart Model ID...")
+                self.model_server = ModelServer.TGI
```
Comment (qiyunz) [inline src/sagemaker/serve/builder/jumpstart_builder.py:502]: why mapping to TGI? should be MMS right?

Comment (makungaj1) [reply src/sagemaker/serve/builder/jumpstart_builder.py]: Fixed

**Review by qiyunz — APPROVED** [review]

**Review by samruds — APPROVED** [review]

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**